### PR TITLE
Fix configuration of test filtering in Android application test runner

### PR DESCRIPTION
**Pull request by elinor-fung**
- Application entry points: configure test runner with current application options
  - Android and iOS application entry points were running the tests and then trying to configure filters for the test run afterwards.
- XUnit test case filtering:
  - Handle different test case filter types separately
  - Use added filters to determine the default for include/exclude
- Add tests for XUnit test case filtering

cc @steveisok @jkoritzinsky

### CODE REVIEW

File: src/Microsoft.DotNet.XHarness.TestRunners.Common/ApplicationEntryPoint.cs
Comment (akoeplinger) [inline src/Microsoft.DotNet.XHarness.TestRunners.Common/ApplicationEntryPoint.cs:8]: it would be nice to avoid the System.Linq dependency

Comment (elinor-fung) [reply src/Microsoft.DotNet.XHarness.TestRunners.Common/ApplicationEntryPoint.cs]: Switched the properties on the internal `ApplicationOptions` to be `ICollection` instead of `IEnumerable` and removed the System.Linq dependency.

**Review by premun — APPROVED** [review]

**Comment by elinor-fung** [comment]
@premun do you think this is ready to merge, or do we need to wait for other reviews? I'd like to get this in soon, since it really helps with the inner dev loop for running Android tests. (I don't have merge permissions)

**Pull request merged**
